### PR TITLE
Add pairwise similarity metrics

### PR DESCRIPTION
**Pull request by iuliaturc**
Adds 4 metrics from the [HEIM benchmark](https://crfm.stanford.edu/helm/heim/latest/) that measure pairwise similarity between images. Useful to gauge progress during training.

### CODE REVIEW

File: eval.py [hunk 1]
```diff
@@ -135,9 +175,13 @@ def read_images(image_dir: str) -> Dict[str, Image.Image]:
...
     images = []
     prompts = []
+
     with open(prompts_path, "r") as f:
         prompts_by_image_filename = json.load(f)
-    for image_filename, prompt in prompts_by_image_filename.items():
+        # It's important to sort the filenames for pairwise similarity metrics.
+        prompts_by_image_filename = sorted(prompts_by_image_filename.items(), key=lambda x: x[0])
+
+    for image_filename, prompt in prompts_by_image_filename:
```
Comment (mihail911) [inline eval.py:184]: didn't realize you could unpack this without explicitly calling out `prompts_by_image_filename.items()`

File: image_eval/pairwise_evaluators.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        if self.normalize:
+            img_transforms.append(transforms.Normalize(mean=(0.5, 0.5, 0.5), std=(0.5, 0.5, 0.5)))
+
+        preprocessing = transforms.Compose(img_transforms)
+        generated_images = [preprocessing(img) for img in generated_images]
+        real_images = [preprocessing(img) for img in real_images]
+
+        generated_images = torch.stack(generated_images).to(self.device)
+        real_images = torch.stack(real_images).to(self.device)
+        logging.info("Mean delta:")
```
Comment (mihail911) [inline image_eval/pairwise_evaluators.py:61]: nit: just use an f-string and then you can combine the two logging.info calls

File: eval.py (hunk 1 above)
Comment (iuliaturc) [reply eval.py:184]: I do on line 182

**Review by mihail911 — APPROVED** [review]
Dope

File: image_eval/pairwise_evaluators.py (hunk 2 above)
Comment (iuliaturc) [reply image_eval/pairwise_evaluators.py:61]: I'll actually remove the logging statements, I left them in by mistake.

**Pull request merged**